ispc-opt.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+  Copyright (c) 2024, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "ispc.h"
+#include "llvmutil.h"
+#include "opt.h"
+#include "src/opt/ISPCPasses.h"
+#include "target_enums.h"
+#include "util.h"
+
+#include "llvm/Support/CommandLine.h"
+#include <llvm/Support/Signals.h>
+#if ISPC_LLVM_VERSION >= ISPC_LLVM_14_0
+#include <llvm/MC/TargetRegistry.h>
+#else
+#include <llvm/Support/TargetRegistry.h>
+#endif
+#if ISPC_LLVM_VERSION >= ISPC_LLVM_16_0
+#include <llvm/IRPrinter/IRPrintingPasses.h>
+#else
+#include <llvm/IR/IRPrintingPasses.h>
+#endif
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IRReader/IRReader.h"
+#include <llvm/Support/TargetSelect.h>
+#include <llvm/Support/ToolOutputFile.h>
+
+using namespace llvm;
+
+static cl::opt<std::string> Passes("passes", cl::desc("Passes to run"));
+static cl::opt<bool> PrintPasses("print-passes", cl::desc("Print passes"));
+static cl::opt<std::string> InputFilename(cl::Positional, cl::desc("<input bitcode file>"), cl::init("-"),
+                                          cl::value_desc("filename"));
+static cl::opt<std::string> OutputFilename("o", cl::desc("Override output filename"), cl::value_desc("filename"));
+static cl::opt<std::string> TargetTarget("target", cl::desc("ISPC target"), cl::init("host"), cl::value_desc("target"));
```

Comment:
`TargetTarget` seems slightly awkward. Why not just `Target`?

Reply:
It conflicts with `Target` class from `llvm/MC/TargetRegistry.h`.